File: Sprint4/D/C++/main.cpp
// https://contest.yandex.ru/contest/23991/problems/D/
#include <iostream>
#include <map>
#include <vector>

int main() {
    int n;
    std::map<std::string, int> map;
    std::cin >> n;

    std::vector<std::string> names(n);
    std::cin.ignore();

    for (int i = 0; i < n; i++) {
        std::string name;
        std::getline(std::cin, name);
        names[i] = name;

        map[name]++;
    }

    for (const auto &name: names) {
        int value = map[name];
        if (value) {
            std::cout << name << std::endl;
            map[name] = 0;
        }
    }

    return 0;
}

File: Sprint4/D/Rust/src/main.rs
// https://contest.yandex.ru/contest/23991/problems/D/
fn main() {
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).unwrap();
    let n: usize = input.trim().parse().unwrap();

    let mut map: std::collections::HashMap<String, i32> = std::collections::HashMap::new();
    let mut names = Vec::new();

    for _ in 0..n {
        input.clear();
        std::io::stdin().read_line(&mut input).unwrap();
        let name = input.trim().to_string();
        *map.entry(name.clone()).or_insert(0) += 1;
        names.push(name);
    }

    for name in &names {
        if let Some(value) = map.get_mut(name) {
            if *value > 0 {
                println!("{}", name);
                *value = 0;
            }
        }
    }
}
